From 66c738b1565fc43851ce9c7122abc2372384d2a0 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Mon, 24 Sep 2018 23:28:05 -0700
Subject: [PATCH 17/18] legacy: Correctly label display.qservice

* Since this is not hosted in a vendor service anymore, this needs to be
  listed as a regular service, not a hwservice.

Change-Id: Icc72d329f534e942c5873e6f7963c2b1072aee2d
---
 common/service.te          | 1 +
 common/service_contexts    | 1 +
 common/surfaceflinger.te   | 2 ++
 common/vndservice.te       | 1 -
 common/vndservice_contexts | 1 -
 5 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/common/service.te b/common/service.te
index 14e9153..afc2b1f 100644
--- a/common/service.te
+++ b/common/service.te
@@ -13,3 +13,4 @@ type dtseagleservice_service,     service_manager_type;
 type gba_auth_service,            service_manager_type;
 type mdtpdaemon_service,          service_manager_type;
 type qtitetherservice_service,    service_manager_type;
+type qdisplay_service,            service_manager_type;
diff --git a/common/service_contexts b/common/service_contexts
index 7cea549..aa7bac4 100644
--- a/common/service_contexts
+++ b/common/service_contexts
@@ -45,3 +45,4 @@ com.qualcomm.qti.uceservice                    u:object_r:imsrcs_service:s0
 # DOLBY_START
 media.dolby_memoryservice                      u:object_r:audioserver_service:s0
 # DOLBY_END
+display.qservice                               u:object_r:qdisplay_service:s0
diff --git a/common/surfaceflinger.te b/common/surfaceflinger.te
index c9a551b..a9f7891 100644
--- a/common/surfaceflinger.te
+++ b/common/surfaceflinger.te
@@ -33,6 +33,8 @@ allow surfaceflinger cameraserver_service:service_manager find;
 #Allow access to binder callback's to camera hal
 binder_call(surfaceflinger, hal_camera_default)
 
+allow surfaceflinger qdisplay_service:service_manager { add find };
+
 #diag
 userdebug_or_eng(`
     diag_use(surfaceflinger)
diff --git a/common/vndservice.te b/common/vndservice.te
index c5360fc..cfacd53 100644
--- a/common/vndservice.te
+++ b/common/vndservice.te
@@ -26,7 +26,6 @@
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 type per_mgr_service,             vndservice_manager_type;
-type qdisplay_service,            vndservice_manager_type;
 type qseeproxy_service,           vndservice_manager_type;
 type esepmdaemon_service,         vndservice_manager_type;
 type wfdnativemm_service,         vndservice_manager_type;
diff --git a/common/vndservice_contexts b/common/vndservice_contexts
index d345755..c704743 100644
--- a/common/vndservice_contexts
+++ b/common/vndservice_contexts
@@ -26,7 +26,6 @@
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 vendor.qcom.PeripheralManager                  u:object_r:per_mgr_service:s0
-display.qservice                               u:object_r:qdisplay_service:s0
 com.qualcomm.qti.qseeproxy                     u:object_r:qseeproxy_service:s0
 eSEPowerManagerService                         u:object_r:esepmdaemon_service:s0
 wfd.native.mm.service                          u:object_r:wfdnativemm_service:s0
-- 
2.17.1

